apps/myaccount/src/hooks/use-sign-in.ts

```diff
@@ -0,0 +1,441 @@
+/**
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {
+    AsgardeoAuthException,
+    AuthenticatedUserInfo,
+    BasicUserInfo,
+    DecodedIDTokenPayload,
+    OIDCEndpoints,
+    useAuthContext
+} from "@asgardeo/auth-react";
+import { AppConstants as AppConstantsCore, CommonConstants as CommonConstantsCore } from "@wso2is/core/constants";
+import {
+    setDeploymentConfigs,
+    setI18nConfigs,
+    setServiceResourceEndpoints,
+    setSignIn,
+    setUIConfigs
+} from "@wso2is/core/store";
+import { AuthenticateUtils, ContextUtils } from "@wso2is/core/utils";
+import { I18nModuleOptionsInterface } from "@wso2is/i18n";
+import { useDispatch } from "react-redux";
+import { AnyAction } from "redux";
+import { ThunkDispatch } from "redux-thunk";
+import useAuthorization from "./use-authorization";
+import useOrganizations from "./use-organizations";
+import { Config } from "../configs/app";
+import { AppConstants } from "../constants/app-constants";
+import { CommonConstants } from "../constants/common-constants";
+import { DeploymentConfigInterface, ServiceResourceEndpointsInterface, UIConfigInterface } from "../models/app-config";
+import { AppState } from "../store";
+import { getProfileInformation, resolveIdpURLSAfterTenantResolves } from "../store/actions/authenticate";
+
+const AUTHORIZATION_ENDPOINT: string = "authorization_endpoint";
+const TOKEN_ENDPOINT: string = "token_endpoint";
+const OIDC_SESSION_IFRAME_ENDPOINT: string = "oidc_session_iframe_endpoint";
+const LOGOUT_URL: string = "sign_out_url";
+
+/**
+ * Props interface of {@link useSignIn}
+ */
+export interface UseSignInInterface {
+    /**
+     * Handles the sign-in process.
+     * @param response - The basic user information returned from the sign-in process.
+     * @returns A promise.
+     */
+    onSignIn: (response: BasicUserInfo) => Promise<void>;
+}
+
+/**
+ * Hook that provides access to the Organizations context.
+ *
+ * @returns An object containing the current Organizations context.
+ */
+const useSignIn = (): UseSignInInterface => {
+    const dispatch: ThunkDispatch<AppState, any, AnyAction> = useDispatch();
```

Comment:
Shall we use type `Dispatch` from redux lib instead?

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/5141/commits/cec412cdc4921989a3ac1867458613650b88e620